documentation/src/main/asciidoc/reference/coordination.asciidoc

```diff
@@ -0,0 +1,508 @@
+and as such they should be automatically generated when relying on Hibernate ORM's
+link:{hibernateDocUrl}#configurations-hbmddl[automatic schema generation].
+
+If you need to integrate the creation/dropping of these tables to your own script,
+the easiest solution is to have Hibernate ORM generate DDL scripts for your whole schema
+and copy everything related to constructs (tables, sequences, ...) prefixed with `HSEARCH_`.
+See link:{hibernateDocUrl}#configurations-hbmddl[automatic schema generation],
+in particular the Hibernate ORM properties `javax.persistence.schema-generation.scripts.action`,
+`javax.persistence.schema-generation.scripts.create-target`
+and `javax.persistence.schema-generation.scripts.drop-target`.
+
+[[coordination-database-polling-sharding]]
+=== Sharding
+
+[[coordination-database-polling-sharding-basics]]
+==== Basics
+
+In order to avoid unnecessarily indexing the same entity multiple times on different application nodes,
+Hibernate Search partitions the entities in what it calls "shards":
+
+* Each entity belongs to exactly one shard.
+* Each application node involved in <<coordination-database-polling-processors,event processing>>
+is uniquely assigned one or more shards, and will only process events related to entities in these shards.
+
+How shards are assigned to application nodes can be customized, as explained in the following sections.
+
+[[coordination-database-polling-sharding-dynamic]]
+==== Dynamic sharding
+
+include::components/dynamic-sharding-not-implemented-warning.asciidoc[]
+
+[[coordination-database-polling-sharding-static]]
+==== Static sharding
+
+Sharding can be controlled explicitly by setting the following configuration properties:
+
+[source]
+----
+hibernate.search.backend.coordination.shards.static = true
+hibernate.search.backend.coordination.shards.total_count = 4
+hibernate.search.backend.coordination.shards.assigned = 0
+----
+
+* `shards.static` defines sharding as static.
+It defaults to `false` and must be set to `true` explicitly in order for
+the other configuration properties to be taken into account.
+* `shards.total_count` defines the total number of shards
+as an <<configuration-property-types,integer value>>.
+This property has no default and must be set explicitly when static sharding is enabled.
+It must be set to the same value on all application nodes with assigned shards.
+* `shards.assigned` defines the shards assigned to the application node
+as an <<configuration-property-types,integer value>>, or multiple comma-separated integer values.
+This property has no default and must be set explicitly when static sharding is enabled.
+Shards are referred to by an index in the range `[0, total_count - 1]` (see above for `total_count`).
+A given application node must be assigned at least one shard
+but may be assigned multiple shards by setting `shards.assigned` to a comma-separated list,
+e.g. `0,4`.
```

Comment:
I wouldn't use the `4` in this example, because even if this is unrelated we said that we had 4 nodes just before. Maybe we should use `3`?